src/workloads/query/PercentilesWindowSpillToDisk.yml

```diff
@@ -0,0 +1,149 @@
+
+Keywords:
+- setWindowFields
+- percentile
+- spill
+
+AutoRun:
+- When:
+    mongodb_setup:
+      $eq:
+      - standalone-all-feature-flags
+      - standalone-classic-query-engine
+      - standalone-sbe
+    branch_name:
+      $neq:
+      - v4.0
+      - v4.2
+      - v4.4
+      - v5.0
+      - v6.0
+
+GlobalDefaults:
+  MaxPhases: &maxPhases 8
+  Database: &db percentiles
+  Collection: &coll Collection0
+
+ActorTemplates:
+- TemplateName: ComputePercentile
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "ComputePercentile"}}
+    Type: CrudActor
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Repeat: 10
+          Database: *db
+          Collection: *coll
+          Operations:
+          - OperationName: aggregate
+            OperationCommand:
+              Pipeline: [{
+                $setWindowFields: {
+                  partitionBy: {^Parameter: {Name: "Partition", Default: null}},
+                  sortBy: {^Parameter: {Name: "Sort", Default: {not_spill_indexed: 1}}},
+                  output: {
+                    p: {
+                      $percentile: {p: [0.75], input: "$data", method: "approximate" },
+                      window: { documents: [0, "unbounded"] }
+                    }}
+                }}
+              ]
+              Options:
+                AllowDiskUse: {^Parameter: {Name: "Disk", Default: true}}
+
+Actors:
+- Name: CreateDataset
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *maxPhases
+      PhaseConfig:
+        Threads: 1
+        Repeat: 1
+        Database: *db
+        CollectionCount: 1
+        # We have 100k documents with each being between 10kB to 20kB. When divided into 20 partitions,
+        # it won't spill, but when divided into fewer partitions (like 2) it will spill.
+        DocumentCount: 100000
+        BatchSize: 1000
+        Document:
+          _id: {^Inc: {start: 0}}
+          # This data field is what we calculate the percentiles of.
+          data: {^RandomDouble: {distribution: normal, mean: 50.0, sigma: 10.0}}
+          # This string pads the doc size out to around 10KB. Since spilling occurs in '$_internalSetWindowFields'
+          # and spills full documents, even though 'string' is not used in the $percentile window function, this
+          # field will increase the document size to trigger a spill.
+          string: &string {^RandomString: {length: 10000}}
+          # We will use the fields below to create approximately even partitions.
+          spill_indexed: {^Choose: {from: [1, 2]}}
+          spill_unindexed: {^Choose: {from: [1, 2]}}
+          not_spill_indexed: {^Choose: {
+            from: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]}}
+
+- Name: CreateIndexes
```

Comment:
I've happened to see in BigUpdate.yml workload that it should be possible to create indexes as part of the loader rather than in a separate phase (which makes sense for us because we don't care to measure perf of index creation)

Reply:
Done!